presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -475,7 +483,23 @@ public CounterStat getHighMemorySplitSource()
 
                 results.add(new HivePartitionMetadata(hivePartition, Optional.of(partition), partitionSchemaDifference.build(), encryptionInformation));
             }
-
+            StringBuilder warningMessage = new StringBuilder(format("Table '%s' has '%s' out of '%s' partitions unreadable: ", tableName, unreadablePartitionsSkipped, partitionBatch.size()));
```

Comment:
In test `testPartitionNotReadable`, right below `assertEquals(SPLIT_SCHEDULING_CONTEXT.getWarningCollector().getWarnings().size(), 1);`, assert the warning you have is the one you expect

Reply:
got it